lib/web/servers.go

```diff
@@ -78,8 +78,27 @@ func (h *Handler) clusterDatabasesGet(w http.ResponseWriter, r *http.Request, p
 		databases = append(databases, server.GetDatabase())
 	}
 
+	accessChecker, err := sctx.GetUserAccessChecker()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	dbNames, dbUsers, err := accessChecker.CheckDatabaseNamesAndUsers(0, true /* force ttl override*/)
+	if err != nil {
+		// if NotFound error:
+		// This user cannot request database access, has no assigned database names or users
+		//
+		// Every other error should be reported upstream.
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err)
+		}
+
+		// We proceed with an empty list of DBUsers and DBNames
+		dbUsers = []string{}
+		dbNames = []string{}
+	}
+
```

Comment:
This code looked familiar :sweat_smile: 

Can we deduplicate this code into a single method and use it for both Reading and Listing Databases endpoints?